File: p171e/title_to_number.cpp
// g++ -std=c++11 *.cpp -o test && ./test && rm -f test

#include <string>
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    int titleToNumber(string s) {
        int ans = 0;
        for (auto c: s) {
            ans = ans * 26 + (c - 'A' + 1) ;
        }
        return ans;
    }
};

int main(int argc, char const *argv[])
{
    vector<string> tests = {
        "A",
        "AB",
        "ZY"
    };
    Solution sol;
    for (auto& t: tests) {
        cout << t << " -> " << sol.titleToNumber(t) << endl;
    }
    return 0;
}

File: p171e/title_to_number.py
class Solution:
    def titleToNumber(self, s: str) -> int:
        ans = 0
        for c in s:
            ans = ans * 26 + ord(c) - ord("A") + 1
        return ans


# TESTS
tests = [
    ("A", 1),
    ("Z", 26),
    ("AA", 27),
    ("AB", 28),
    ("ZY", 701),
]
for t in tests:
    sol = Solution()
    actual = sol.titleToNumber(t[0])
    print("Column title of", t[0], "->", actual)
    assert actual == t[1]
